### DISCUSSION TITLE
MQTT Broker Connection with Microcks on WSL

### DISCUSSION THREAD

**Discussion by jonase97** [microcks/microcks#1383] [Q&A]
Hello everybody,

I'm currently trying to get Microcks running locally on a Windows 10 laptop. I have it installed under WSL2 with Docker Compose. The web interface is available through the browser running on localhost:8080. I was also able to login, upload an AsyncAPI document and create tests for the operations. 
However, there seems to be a problem connecting to the mqtt broker (ActiveMQ) which is running on a different docker container and available on localhost:61616. The broker is definitely available, for example through Postman or other client applications.
It appears as if Microcks can't connect to the broker, there aren't any messages coming through even though the Mocks are enabled and when I try to run tests, they terminate immediately even though there should be at least a 10 second timeout.

![microcks_operation](https://github.com/user-attachments/assets/62c7f709-56eb-4ab5-b575-2a9258a01867)

![microcks_test](https://github.com/user-attachments/assets/812f581b-6114-4bdb-956f-c6739cbec67b)

As mentioned in other discussion topics, I have adjusted the application.properties and feature.properties in the install/docker-compose folder.

![applciation_properties](https://github.com/user-attachments/assets/74d26a77-6c92-44e7-9d12-ecd9f1ad7881)

![feature_properties](https://github.com/user-attachments/assets/4c29c907-273b-48f0-9067-476b3912fdb0)

I'm not sure if I'm missing something obvious, but currently I don't even know where to look for the cause of the error and I was hoping someone has an idea what the problem could be.

Thanks in advance!

**Comment by lbroudoux** [comment]
Hey @jonase97 

This is not an obvious thing, but these are the joys of container networking! `localhost` from the Microcks container point of view is resolved to itself but here you want to join your Windows 10/WSL host. Most of the time, your host can be joined using `host.docker.internal` as an alias. So you must replace `localhost:61616` with `host.docker.internal:61616`.

The other way of doing that is to start ActiveMQ in the same docker network and use the container name as the network alias. That's typically what we're doing for Microcks to Kafka communication. See https://github.com/microcks/microcks/blob/master/install/docker-compose/docker-compose-async-addon.yml#L57

Hope it helps!
